services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/common/AttachmentContentEntity.kt

```diff
@@ -0,0 +1,48 @@
+package fi.hel.haitaton.hanke.attachment.common
+
+import java.util.UUID
+import javax.persistence.Basic
+import javax.persistence.Column
+import javax.persistence.Entity
+import javax.persistence.FetchType
+import javax.persistence.Id
+import javax.persistence.Lob
+import javax.persistence.MappedSuperclass
+import javax.persistence.Table
+import javax.validation.constraints.NotNull
+import org.hibernate.annotations.Type
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.stereotype.Repository
+
+@MappedSuperclass
+abstract class AttachmentContentEntity(
+    @Id @Column(name = "attachment_id") var attachmentId: UUID,
+
+    /** Attachment data, i.e. the file itself. */
+    @Lob
+    @Type(type = "org.hibernate.type.BinaryType")
+    @Basic(fetch = FetchType.LAZY)
```

Comment:
I think Basic(fetch = FetchType.LAZY) was not working in the first place. It could possibly be removed.

See: https://vladmihalcea.com/the-best-way-to-lazy-load-entity-attributes-using-jpa-and-hibernate/

Basic(fetch = FetchType.LAZY)

This configuration alone is not sufficient because Hibernate requires bytecode instrumentation to intercept the attribute access request and issue the secondary select statement on demand.</p>